alarm/model.py

```diff
@@ -0,0 +1,170 @@
+"""The model of an alarm
+---------------------
+
+About this Project
+------------------
+The objective of this project is to recreate an alarm that a user could set
+interactively and which will ring back at this preset time.
+
+Project structure
+-----------------
+*alarm/*
+    **__main__.py**:
+        The application of an alarm
+    **alarm_test.py**:
+        The tests of an alarm
+    **controller.py**:
+        The controller of an alarm
+    **-->** **model.py**:
+        The model of an alarm
+    **view.py**:
+        The GUI of an alarm
+
+About this module
+-----------------
+The objective of this module is to model an alarm using Python. It uses
+innovative Python libraries such as logging, time, datetime and subprocess
+which helped to build the application using the current date and time as well
+as to provide a user interface to set the alarm according to the requirement in
+24-hour format.
+
+File structure
+--------------
+*import*
+    **logging**
+        provides a logging tool to inform the user of the system state through
+        the console.
+    **time, datetime**
+        helps working with the time of the current day.
+    **subprocess**
+        provides access to a basic sound playing machinery through the Linux
+        system with ffmpeg (ffplay). This was useful to generate the sound
+        immediately when the function is called and not waiting for the
+        response (async).
+
+*constant*
+    **FFPLAY_CMD**
+        hardcodes the command to avoid that launching the ffplay GUI hides text
+        printout to console. It also hardcodes the filename to play.
+    **MSG_***
+        string messages usable by the logger at one point in the code.
+    **TIME_INTERVAL**
+        keeps the clock polling the alarm set by the user for every 1 second,
+        the string format/pattern for the alarm.
+    **TIME_FORMAT, TIME_PATTERN**
+        useful to compare alarm with actual time.
+
+*class*
+    **Alarm**
+        'The abstraction of an alarm'
+    **alarm_time(self)**
+        'Get the current alarm time'
+    **is_alive(self)**
+        'Determine if the alarm should continue to run or stop'
+    **manage_time(self, now)**
+        'Manage what the alarm should do between ringing and sleeping'
+    **ring(self)**
+        'Play the sound WAV_FILENAME using the ffplay command'
+    **run(self)**
+        'Define the main thread that handle the alarm'
+"""
+import logging
+import time
+from collections import namedtuple
+from datetime import datetime
+from subprocess import Popen, DEVNULL
+
+# Sound play command constants
+FFPLAY_CMD = ['ffplay', '-nodisp', '-autoexit', '-hide_banner', '-v', '0', '/usr/share/sounds/purple/login.wav']
+
+# Logging messages
+MSG_INFO_START = "Starting alarm"
+MSG_INFO_STOP = "Stopping alarm"
+MSG_DEBUG_SLEEP = "self.alarm_time={}, self.sleep_time={}"
+MSG_INFO_WAKEUP = "Wake up Dude"
+
+# Time related
+ATIME = namedtuple('at', ['hour', 'min', 'sec'])
```

Comment:
Does 'at' is in fact 'aTime', to be more consistent?

Reply:
I put ATIME in all caps as it is a constant.